File: src/main/kotlin/com/itmo/microservices/demo/order/impl/entity/CatalogItem.java
package com.itmo.microservices.demo.order.impl.entity;

import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import lombok.ToString;
import org.hibernate.Hibernate;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import java.util.Objects;
import java.util.UUID;

@Entity
@Getter
@Setter
@ToString
@RequiredArgsConstructor
public class CatalogItem {
    @Id
    @GeneratedValue
    private UUID uuid;

    private String title;
    private String description;
    private int price;
    private int amount;

    @ManyToOne
    private Order order;

    public CatalogItem(UUID randomUUID, String title, String description, int price, int amount) {
        this.uuid = randomUUID;
        this.title = title;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || Hibernate.getClass(this) != Hibernate.getClass(o)) return false;
        CatalogItem that = (CatalogItem) o;
        return uuid != null && Objects.equals(uuid, that.uuid);
    }

    @Override
    public int hashCode() {
        return getClass().hashCode();
    }
}

File: src/main/kotlin/com/itmo/microservices/demo/warehouse/impl/entity/CatalogItem.kt
package com.itmo.microservices.demo.warehouse.impl.entity

import java.util.UUID
import com.fasterxml.jackson.annotation.JsonBackReference
import javax.persistence.*
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull

@Entity
@Table(name = "catalog_items")
class CatalogItem {
    @Id
    @GeneratedValue
    var id: UUID? = null

    @JsonBackReference
    @OneToOne(cascade = [CascadeType.REMOVE], fetch = FetchType.EAGER)
    @PrimaryKeyJoinColumn
    var warehouseItem: WarehouseItem? = null

    @field:NotNull(message = "Title must be not null")
    var title: String? = null

    @field:NotNull(message = "Description must be not null")
    var description: String? = null

    @field:NotNull(message = "Price must be not null")
    @field:Min(value = 1, message = "Price must be greater than 0")
    var price = 100

    constructor() {}
    constructor(
        id: UUID?,
        warehouseItem: WarehouseItem?,
        title: String?,
        description: String?,
        price: Int
    ) {
        this.id = id
        this.warehouseItem = warehouseItem
        this.title = title
        this.description = description
        this.price = price
    }
}